src/Parser/VengeanceDemonHunter/CombatLogParser.js

```diff
@@ -0,0 +1,191 @@
+import StatisticBox from 'Main/StatisticBox';
+import SuggestionsTab from 'Main/SuggestionsTab';
+import Tab from 'Main/Tab';
+import Talents from 'Main/Talents';
+import getCastEfficiency from 'Parser/Core/getCastEfficiency';
+
+import ISSUE_IMPORTANCE from 'Parser/Core/ISSUE_IMPORTANCE';
+
+import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
+
+import CPM_ABILITIES from './CPM_ABILITIES';
+
+
+function formatThousands(number) {
+  return (Math.round(number || 0) + '').replace(/(\d)(?=(\d{3})+(?!\d))/g, "$1,");
+}
+
+function formatNumber(number) {
+  if (number > 1000000) {
+    return `${(number / 1000000).toFixed(2)}m`;
+  }
+  if (number > 10000) {
+    return `${Math.round(number / 1000)}k`;
+  }
+  return formatThousands(number);
+}
+
+function formatPercentage(percentage) {
+  return (Math.round((percentage || 0) * 10000) / 100).toFixed(2);
+}
+
+function getIssueImportance(value, regular, major, higherIsWorse = false) {
+  if (higherIsWorse ? value > major : value < major) {
+    return ISSUE_IMPORTANCE.MAJOR;
+  }
+  if (higherIsWorse ? value > regular : value < regular) {
+    return ISSUE_IMPORTANCE.REGULAR;
+  }
+  return ISSUE_IMPORTANCE.MINOR;
+}
+
+const damageType = {
+  1: 'Physical',
+  2: 'Holy',
+  4: 'Fire',
+  8: 'Nature',
+  16: 'Frost',
+  28: 'Elemental',
+  32: 'Shadow',
+  33: 'Shadowstrike',
+  34: 'Twilight',
+  36: 'Shadowflame',
+  40: 'Plague',
+  48: 'Shadowfrost',
+  64: 'Arcane',
+  96: 'Spellshadow',
+  100: 'Special',
+  124: 'Chaos',
+};
+
+function getMagicDescription(type) {
+  if (damageType[type] === undefined) {
+    return 'Chaos';
+  }
+  return damageType[type];
+}
+
+class CombatLogParser extends MainCombatLogParser {
+  static specModules = {
+    // Features
+    alwaysBeCasting: AlwaysBeCasting,
+  };
+
+  damageBySchool = {};
+  on_toPlayer_damage(event) {
+    if (this.damageBySchool[event.ability.type] === undefined) {
+      this.damageBySchool[event.ability.type] = 0;
+    }
+    this.damageBySchool[event.ability.type] += event.amount + (event.absorbed || 0);
+    super.on_toPlayer_damage(event);
+  }
+
+  generateResults() {
+    const results = super.generateResults();
+
+    const fightDuration = this.fightDuration;
+
+    const nonDpsTimePercentage = this.modules.alwaysBeCasting.totalDamagingTimeWasted / fightDuration;
+    const deadTimePercentage = this.modules.alwaysBeCasting.totalTimeWasted / fightDuration;
+
+    console.log(this.modules.alwaysBeCasting.totalTimeWasted+'---'+fightDuration);
+
+    if (nonDpsTimePercentage > 0.01) {
+      results.addIssue({
+        issue: `Your non DPS time can be improved. Try to cast damaging spells more regularly.`,
+        stat: `${Math.round(nonDpsTimePercentage * 100)}% non DPS time (<30% is recommended)`,
+        icon: 'petbattle_health-down',
+        importance: getIssueImportance(nonDpsTimePercentage, 0.4, 0.45, true),
+      });
+    }
+    if (deadTimePercentage > 0.2) {
+      results.addIssue({
```

Comment:
With the new module setup I believe this belong to the AlwaysBeCasting module; in the `suggestions(when)` method